### ISSUE TITLE
Include more deadkeys

### ISSUE THREAD

**Issue by Meriipu** [legionus/kbd#43]
Is there some way for the user to define a key to be a deadkey?
As an example, I tried to make a key output `dead_abovering`, but it did not seem to be defined. 
It was relatively recently included in the kernel
https://github.com/torvalds/linux/commit/16777ecd1b54d75136f77b2cc25f2cfa75156852

If there is not support for it, is there a workaround that does not involve pressing a compose-key first? 
A compose key requires three keypresses from what I can tell, rather than the two that a single deadkey requires. 

I attempted to add `dead_abovering` in myself, though there is some behaviour I do not really understand:
```
---
 src/libkeymap/syms.ktyp.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/libkeymap/syms.ktyp.h b/src/libkeymap/syms.ktyp.h
index b319388..1b43975 100644
--- a/src/libkeymap/syms.ktyp.h
+++ b/src/libkeymap/syms.ktyp.h
@@ -386,7 +386,8 @@ static const char *const dead_syms[] = {
 	"dead_tilde",
 	"dead_diaeresis",
 	"dead_cedilla",
-	"dead_macron"
+	"dead_macron",
+	"dead_abovering"
 };
 
 /*
-- 
2.27.0
```

I made a key output `dead_abovering` added two compose-rules at the bottom of my keymap (the documentation is not too clear on how that works for diacritics/deadkeys...):
```
keycode  41 = dead_abovering             dead_diaeresis          grave

compose 'U' 'a' to 0xe5
compose 'U' 'A' to 0xc5
```
The part that is strange to me is that the diacritic appears as `U` if there is no rule to compose it with. 
Where does `U` come from? I would have thought it would use either `degree`, or the `*` from the kernel table.
The keymap works, though, and I can use abovering as a deadkey with this.

Something which added to my confusion was `docs/doc/kbd.FAQ.txt` (The Linux keyboard and console HOWTO):
```
  +^Ho  A dead diacritic followed by a symbol, yields that symbol adorned
     with that diacritic. If the combination is undefined, both keys are
     taken separately.  Which keys are dead diacritics is user-settable;
     none is by default.  Five (since 2.0.25 six) dead diacritics can be
     defined (using loadkeys(1)): dead_grave, dead_acute,
     dead_circumflex, dead_tilde, dead_diaeresis (and dead_cedilla).
     Precisely what this adorning means is also user-settable: dead-
     diacritic, symbol is equivalent to Compose + diacritic + symbol.
```
maybe this document is very dated.

**Comment by Meriipu** [comment]
Oh wait, I think I understand the `U` now. It is the 8th entry in the kernel table, immediately following the 7 that are understood by kbd (and the entry I added to kbd is at that same index), yes?

**Comment by Meriipu** [comment]
Is this more or less how to?
```
From 298ef3f1eed631aa5d427acee466f24ea901d491 Mon Sep 17 00:00:00 2001
From: Meriipu <Meriipu@users.noreply.github.com>
Date: Mon, 27 Jul 2020 03:23:22 +0200
Subject: [PATCH] Additional deadkeys which have been in the kernel for around
 2 years

https://lore.kernel.org/patchwork/patch/912759/
  '`',  /* dead_grave */
  '\'', /* dead_acute */
  '^',  /* dead_circumflex */
  '~',  /* dead_tilda */
  '"',  /* dead_diaeresis */
  ',',  /* dead_cedilla */
  '_',  /* dead_macron */
  'U',  /* dead_breve */
  '.',  /* dead_abovedot */
  '*',  /* dead_abovering */
  '=',  /* dead_doubleacute */
  'c',  /* dead_caron */
  'k',  /* dead_ogonek */
  'i',  /* dead_iota */
  '#',  /* dead_voiced_sound */
  'o',  /* dead_semivoiced_sound */
  '!',  /* dead_belowdot */
  '?',  /* dead_hook */
  '+',  /* dead_horn */
  '-',  /* dead_stroke */
  ')',  /* dead_abovecomma */
  '(',  /* dead_abovereversedcomma */
  ':',  /* dead_doublegrave */
  'n',  /* dead_invertedbreve */
  ';',  /* dead_belowcomma */
  '$',  /* dead_currency */
  '@',  /* dead_greek */
---
 src/libkeymap/syms.ktyp.h     | 22 +++++++++++++++++++++-
 src/libkeymap/syms.synonyms.h |  8 --------
 2 files changed, 21 insertions(+), 9 deletions(-)

diff --git a/src/libkeymap/syms.ktyp.h b/src/libkeymap/syms.ktyp.h
index b319388..aae6458 100644
--- a/src/libkeymap/syms.ktyp.h
+++ b/src/libkeymap/syms.ktyp.h
@@ -386,7 +386,27 @@ static const char *const dead_syms[] = {
 	"dead_tilde",
 	"dead_diaeresis",
 	"dead_cedilla",
-	"dead_macron"
+	"dead_macron",
+	"dead_breve",
+	"dead_abovedot",
+	"dead_abovering",
+	"dead_doubleacute",
+	"dead_caron",
+	"dead_ogonek",
+	"dead_iota",
+	"dead_voiced_sound",
+	"dead_semivoiced_sound",
+	"dead_belowdot",
+	"dead_hook",
+	"dead_horn",
+	"dead_stroke",
+	"dead_abovecomma",
+	"dead_abovereversedcomma",
+	"dead_doublegrave",
+	"dead_invertedbreve",
+	"dead_belowcomma",
+	"dead_currency",
+	"dead_greek"
 };
 
 /*
diff --git a/src/libkeymap/syms.synonyms.h b/src/libkeymap/syms.synonyms.h
index 7ce4987..a34ef94 100644
--- a/src/libkeymap/syms.synonyms.h
+++ b/src/libkeymap/syms.synonyms.h
@@ -38,14 +38,6 @@ struct syn const synonyms[] = {
 	/* diacriticals */
 	{ "tilde", "asciitilde" },
 	{ "circumflex", "asciicircum" },
-	/* as dead_ogonek, dead_caron, dead_breve and dead_doubleacute do not exist
-	 * (yet), I put also compose lines for use with respectively dead_cedilla,
-	 * dead_circumflex, dead_tilde and dead_tilde
-	 */
-	{ "dead_ogonek", "dead_cedilla" },
-	{ "dead_caron", "dead_circumflex" },
-	{ "dead_breve", "dead_tilde" },
-	{ "dead_doubleacute", "dead_tilde" },
 	/* turkish */
 	{ "Idotabove", "Iabovedot" },
 	{ "dotlessi", "idotless" },
-- 
2.27.0
```

**Comment by legionus** [comment]
> Is this more or less how to?

Yep

**Comment by Meriipu** [comment]
Closed in df4de8f1245fab5395e6ee7ce0ce533c21703366
